lib/Parse/ASTGen.cpp

```diff
@@ -1267,6 +1382,22 @@ SourceLoc ASTGen::advanceLocBegin(const SourceLoc &Loc, const Syntax &Node) {
   return Loc.getAdvancedLoc(Node.getAbsolutePosition().getOffset());
 }
 
+SourceLoc ASTGen::advanceLocEnd(const SourceLoc &Loc, const Syntax &Node) {
+  if (!Node.isMissing()) {
+    // NOTE: We cannot use 'getLastToken()' because it doesn't take string
+    // literal expressions into account.
+    if (Node.isToken() || Node.is<StringLiteralExprSyntax>())
```

Comment:
Since `StringLiteralExprSyntax` is like `{ quote (literal-segment | interpolation-segment)* quote }`,
This used to return the location of the end quote. In this case, we have to return the location of the start quote instead.